### PR TITLE
[log4cplus ] Normilize log4cplus unicode build

### PR DESCRIPTION
**Pull request by ngrodzitski**
Add unicode as a separate feature for log4cplus package and unify target name for windows and linux. Previously on linux non-unicode version was installed, while on windows it was a unicode version. Other issue was that target names were different. On linux a static library is produced and target name is prefixed with "S", while on windows a shared library is produced and prefix is not used - to have the same target name on both environments I disable `LOG4CPLUS_ENABLE_DECORATED_LIBRARY_NAME` option.

- Seems like all triplets are affected. No updates to CI baseline.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

File: ports/log4cplus/portfile.cmake [hunk 1]
```diff
@@ -23,13 +23,21 @@ file(
     DESTINATION ${SOURCE_PATH}/threadpool
 )
 
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        unicode     UNICODE
```
Comment (NancyLi1013) [inline ports/log4cplus/portfile.cmake:29]:
```suggestion
    FEATURES
    unicode     UNICODE
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Have you tested the feature? Does it work fine?

**Comment by ngrodzitski** [comment]
> Have you tested the feature? Does it work fine?

Yes, at least on linux. I'll check on windows later.

**Comment by ngrodzitski** [comment]
Checked on windows - all is OK.

**Comment by NancyLi1013** [comment]
Thanks for your feedback. 
LGTM now.

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**